lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenLemminxExtension.java

```diff
@@ -235,14 +246,52 @@ private MavenExecutionRequest initMavenRequest(PlexusContainer container, XMLMav
 		mavenRequest.setWorkspaceReader(new MavenLemminxWorkspaceReader(this));
 		return mavenRequest;
 	}
+
+	public Settings buildSettings(PlexusContainer container, File globalSettingsFile, File userSettingsFile) {
+		SettingsBuildingRequest request = new DefaultSettingsBuildingRequest();
+		request.setGlobalSettingsFile(globalSettingsFile);
+		request.setUserSettingsFile(userSettingsFile != null ? userSettingsFile
+				: SettingsXmlConfigurationProcessor.DEFAULT_USER_SETTINGS_FILE);
+		try {
+			return container.lookup(SettingsBuilder.class).build(request).getEffectiveSettings();
+		} catch (SettingsBuildingException | ComponentLookupException ex) {
+			return null;
+		}
+	}
 	
+	/**
+	 * Thread-safe properties copy implementation.
+	 * <p>
+	 * {@link Properties#entrySet()} iterator is not thread safe and fails with
+	 * {@link ConcurrentModificationException} if the source properties "is
+	 * structurally modified at any time after the iterator is created". The
+	 * solution is to use thread-safe {@link Properties#stringPropertyNames()}
+	 * enumerate and copy properties.
+	 *
+	 * @see https://bugs.eclipse.org/bugs/show_bug.cgi?id=440696
+	 */
+	public Properties getSystemProperties() {

```

Comment:
I don't think this would have to be public, nor that a field is required, if we use `mavenRequest.getProperties()` later. This could be a `private static` method.

Reply:
ACK